// THIS FILE IS AUTOMATICALLY GENERATED by https://github.com/swift-aws/aws-sdk-swift/blob/master/CodeGenerator/Sources/CodeGenerator/main.swift. DO NOT EDIT.

import AWSSDKSwiftCore

/// Error enum for Appflow
public enum AppflowErrorType: AWSErrorType {
    case conflictException(message: String?)
    case connectorAuthenticationException(message: String?)
    case connectorServerException(message: String?)
    case internalServerException(message: String?)
    case resourceNotFoundException(message: String?)
    case serviceQuotaExceededException(message: String?)
    case unsupportedOperationException(message: String?)
    case validationException(message: String?)
}

extension AppflowErrorType {
    public init?(errorCode: String, message: String?){
        var errorCode = errorCode
        if let index = errorCode.firstIndex(of: "#") {
            errorCode = String(errorCode[errorCode.index(index, offsetBy: 1)...])
        }
        switch errorCode {
        case "ConflictException":
            self = .conflictException(message: message)
        case "ConnectorAuthenticationException":
            self = .connectorAuthenticationException(message: message)
        case "ConnectorServerException":
            self = .connectorServerException(message: message)
        case "InternalServerException":
            self = .internalServerException(message: message)
        case "ResourceNotFoundException":
            self = .resourceNotFoundException(message: message)
        case "ServiceQuotaExceededException":
            self = .serviceQuotaExceededException(message: message)
        case "UnsupportedOperationException":
            self = .unsupportedOperationException(message: message)
        case "ValidationException":
            self = .validationException(message: message)
        default:
            return nil
        }
    }
}

extension AppflowErrorType : CustomStringConvertible {
    public var description : String {
        switch self {
        case .conflictException(let message):
            return "ConflictException: \(message ?? "")"
        case .connectorAuthenticationException(let message):
            return "ConnectorAuthenticationException: \(message ?? "")"
        case .connectorServerException(let message):
            return "ConnectorServerException: \(message ?? "")"
        case .internalServerException(let message):
            return "InternalServerException: \(message ?? "")"
        case .resourceNotFoundException(let message):
            return "ResourceNotFoundException: \(message ?? "")"
        case .serviceQuotaExceededException(let message):
            return "ServiceQuotaExceededException: \(message ?? "")"
        case .unsupportedOperationException(let message):
            return "UnsupportedOperationException: \(message ?? "")"
        case .validationException(let message):
            return "ValidationException: \(message ?? "")"
        }
    }
}
